extensions/wikia/AdEngine/js/slot/premiumFloatingMedrec.js

```diff
@@ -0,0 +1,158 @@
+/*global define Promise*/
+define('ext.wikia.adEngine.slot.premiumFloatingMedrec', [
+	'ext.wikia.adEngine.adContext',
+	'ext.wikia.adEngine.context.uapContext',
+	'ext.wikia.adEngine.slot.service.viewabilityHandler',
+	'wikia.document',
+	'wikia.log',
+	'wikia.throttle',
+	'wikia.window'
+], function (
+	adContext,
+	uapContext,
+	viewabilityHandler,
+	doc,
+	log,
+	throttle,
+	win
+) {
+	'use strict';
+	var logGroup = 'ext.wikia.adEngine.slot.premiumFloatingMedrec';
+
+	function init() {
+		var adSlot = doc.createElement('div'),
+			context = adContext.getContext(),
+			maxChanges = 6,
+			onScroll,
+			placeHolderElement = doc.getElementById('WikiaAdInContentPlaceHolder'),
+			recirculationElement = doc.getElementById('recirculation-rail'),
+			refreshInfo = {
+				refreshAdPos: 0,
+				lastRefreshTime: new Date(),
+				refreshNumber: 0,
+				adVisible: true,
+				heightScrolledThreshold: 10,
+				refreshDelay: 10000
+			},
+			slotName = 'INCONTENT_BOXAD_1',
+			startPosition = placeHolderElement.offsetTop,
+			swapRecirculationAndAd;
+
+		adSlot.className = 'wikia-ad';
+		adSlot.setAttribute('id', slotName);
+
+		function hasUserScrolledEnoughDistance(currentHeightPosition) {
+			var heightScrolled = Math.abs(currentHeightPosition - refreshInfo.refreshAdPos);
+			return heightScrolled > refreshInfo.heightScrolledThreshold;
+		}
+
+		function wasItEnoughTimeSinceLastRefresh() {
+			var timeDifference = (new Date()) - refreshInfo.lastRefreshTime;
+			return timeDifference > refreshInfo.refreshDelay;
+		}
+
+		function isMaxNumberOfRefreshReached() {
+			return refreshInfo.refreshNumber >= maxChanges;
+		}
+
+		function isUAPFloatingMedrecVisible() {
+			var isAdVisible = refreshInfo.adVisible && refreshInfo.refreshNumber !== 0;
+			return uapContext.isUapLoaded() && isAdVisible;
+		}
+
+		function shouldSwitchModules(currentHeightPosition) {
+			return hasUserScrolledEnoughDistance(currentHeightPosition) &&
+				wasItEnoughTimeSinceLastRefresh() &&
+				!isMaxNumberOfRefreshReached() &&
+				!isUAPFloatingMedrecVisible();
+		}
+
+		function updateModulesRefreshInformation(currentHeightPosition) {
+			refreshInfo.lastRefreshTime = new Date();
+			refreshInfo.refreshAdPos = currentHeightPosition;
+			refreshInfo.refreshNumber++;
+		}
+
+		function showRecirculation() {
+			recirculationElement.style.display = 'block';
+		}
+
+		function hideRecirculation() {
+			recirculationElement.style.display = 'none';
+		}
+
+		function refreshAd() {
+			return new Promise(function (resolve, reject) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused function parameter "reject". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)